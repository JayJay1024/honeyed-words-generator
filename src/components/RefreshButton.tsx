import React from 'react';
import styled from 'styled-components';
import ImageRefresh from '../assets/imgs/refresh.svg';
import StyledButton from './StyledButton';

const Button = styled<any>(StyledButton)`
    position: fixed;
    left: 3rem;
    bottom: 0.5rem;
    background-image: url(${ImageRefresh});
`;

const RefreshButton: React.FunctionComponent<{visible: boolean, handleUpdate: any}> = ({ visible = false, handleUpdate }): JSX.Element => {
    return <Button className={visible ? 'visible' : 'hidden'} onClick={handleUpdate}></Button>;
};

export default RefreshButton;
