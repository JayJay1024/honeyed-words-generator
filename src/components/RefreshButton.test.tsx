import React from 'react';
import { create, act, ReactTestRenderer } from 'react-test-renderer';

import RefreshButton from './RefreshButton';

describe('component RefreshButton', () => {
    let testRenderer: ReactTestRenderer = null;

    beforeEach(() => {
        act(() => {
            testRenderer = create(<RefreshButton visible={false} handleUpdate={null} />);
        });
    });

    afterEach(() => {
        if (testRenderer) {
            testRenderer.unmount();
            testRenderer = null;
        }
    });

    test('RefreshButton should match snapshot', () => {
        let tree = testRenderer.toJSON();
        expect(tree).toMatchSnapshot();

        act(() => {
            testRenderer.update(<RefreshButton visible={true} handleUpdate={null} />);
        });
        tree = testRenderer.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('there should be a button', () => {
        const testInstance = testRenderer.root;
        expect(testInstance.findAllByType("button").length).toEqual(1);
    });
});
