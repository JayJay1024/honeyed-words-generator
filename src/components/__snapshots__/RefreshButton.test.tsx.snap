// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component RefreshButton RefreshButton should match snapshot 1`] = `
<button
  className="sc-AxjAm sc-AxirZ kQdOTh hidden"
  onClick={null}
/>
`;

exports[`component RefreshButton RefreshButton should match snapshot 2`] = `
<button
  className="sc-AxjAm sc-AxirZ kQdOTh visible"
  onClick={null}
/>
`;
